roles/dcos_bootstrap/tasks/main.yml

```diff
@@ -280,13 +280,38 @@
     creates: "{{ download_path }}/genconf/serve/windows/upgrade_from_{{ item }}/"
   when:
     - "dcos['version'] is version('1.9', '>=')"
-    - "dcos['download_win'] is defined"
+    - dcos['download_win'] is defined
   loop: >
     {{ groups['dcos'] | default([]) | map('extract',hostvars,'ansible_local')| list |
      select('defined') | selectattr('dcos_installation', 'defined') |
      map(attribute='dcos_installation') | map(attribute='version') | list |
      union([dcos['version_to_upgrade_from'] | default(dcos['version']) ]) | unique }}
 
+- name: Get upgrade directory hash
+  shell: |
+    set -o pipefail
+    ls -td -- */ | head -n 1 | cut -d'/' -f1
+  args:
+    chdir: "{{ download_path }}/genconf/serve/windows/upgrade_from_{{ item }}"
+  changed_when: false
+  register: upgrade_win_dir_hash
+  when: "dcos['version'] is version('1.9', '>=')"
```

Comment:
nope... definitly not... this is more 2.1 and ONLY when a win artifact is given correct?

Reply:
fixed at 83f5b6b93dea38bb8abebb4c3420484eca721c62